### ISSUE TITLE
BarChart with time xScale fails

### ISSUE THREAD

**Issue by Guibod** [react-d3/react-d3-basics#29]
Bar relies on `Xscale.bandwidth()` that is not available in a [Time scale](https://github.com/d3/d3-scale#time-scales).

**Comment by chilijung** [comment]
Hi, @Guibod, why are you using time scale while you are using Bar chart? I'm suppose that passing an ordinal scale in Xscale. 

see example here:

https://github.com/react-d3/react-d3-shape/blob/master/example/src/bar.jsx#L26

**Comment by Guibod** [comment]
From my beginner point of view it sounded pretty smart to use a Time scale in an histogram, yet it was stupid in the logic universe.
Programmatically speaking it is still a possibility, you may need to assert scale type in the API since some unaware people (like myself) don't understand what they are doing sometimes. :)

**Comment by chilijung** [comment]
Yeah, your right! Actually I've been really busy these days and there are many documents that haven't arrange yet. I'll do it when I have time. Thanks.
